### PR TITLE
Fix various problems with bundle links in search results.

### PR DESCRIPTION
**Pull request by kadams54**

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

1. Search for something.
2. Click on a bundle.
3. Verify that the details load OK.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5061/
Test PASSed.

**Comment by jaycee** [comment]
:+1:

**Comment by bac** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5062/
Test PASSed.

File: jujugui/static/gui/src/app/components/search-results/test-search-results.js [hunk 1]
```diff
@@ -623,6 +623,35 @@ describe('SearchResults', function() {
...
         id: 'mysql',
         type: 'charm'
+      }, {
+        name: 'wordpress',
+        displayName: 'wordpress',
+        url: 'http://example.com/wordpress',
+        downloads: 1000,
+        owner: 'charmers',
+        promulgated: true,
+        id: '~charmers/wordpress',
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/test-search-results.js]: Can we use real id's for these? This is kind of why we ended up with this bug: 

"cs:bundle/mediawiki-single-9"

"cs:trusty/rabbitmq-server-43"

for example

**Comment by bac** [comment]
QA: repeated failed attempt. Searched for 'ceph' and examined the four bundles.  All worked this time. Great!

As long as you don't touch the real code when fixing the tests, this is QA-OK.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**